starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,82 @@
-// Names and Input
+// Names and input
 
+var hacker1 = "Mehdi";
+console.log("the driver's name is " + hacker1);
+var hacker2 = prompt("what is passenger's name ?");
+console.log("The navigator name is "  + hacker2);
 
-//Conditionals
+// Conditionals
 
+if (hacker1.length > hacker2.length) {
+  console.log("the driver has the longest name, it has " + hacker1.length + " characters");
+}
 
-// Lorem ipsum generator
+else if (hacker1.length < hacker2.length) {
+  console.log("Yo, navigator got the longest name, it has " + hacker2.length + " characters");
+}
+
+else {
+  console.log("wow, you both got equally long names, " + hacker1.length + " characters!!");
+}
+
+// LOOPS
+
+var out = "";
+for (var i=0; i<hacker1.length; i++){
+  out += (hacker1[i].toUpperCase() + " ");
+};
+console.log(out);
+
+var out = "";
```

Comment:
Be careful not to declare twice variables with the same name, this can lead to bugs in the code !!